prereise/gather/demanddata/transportation_electrification/charging.py

```
@@ -0,0 +1,114 @@
+
+    dwelling = int(trip_data["Dwell time (hour decimal)"] > 0.2)
+    location = int(get_location(location_strategy, trip_data["why to"]))
+
+    trip = int(
+        consider_trip_number(
+            trip_strategy, trip_data["total vehicle trips"], trip_data["trip number"]
+        )
+    )
+
+    battery = int(get_battery_SOC(trip_data["trip end battery charge"], kwh))
+
+    if (dwelling * location * trip * battery != 1) or not (
+        (dwelling * location == 1)
+        or (trip_data["total vehicle trips"] == trip_data["trip number"])
+    ):
+        charging_power = 0
+
+    return charging_power
+
+
+def get_charging_time(charging_power, kwh, battery_SOC, charging_efficiency, trip_data):
+    """Calculates time needed to charge.
+
+    :param float charging_power: charging power.
+    :param float kwh: kwhmi * veh_range, amount of energy needed to charge vehicle.
+    :param float battery_SOC: battery charge at end of trip.
+    :param float charging_efficiency: grid to battery efficiency.
+    :return: (*float*) -- charging time in decimal format.
+    """
+    if charging_power == 0:
+        charging_time = 0
+    else:
+        charging_time = (kwh - battery_SOC) / (charging_power * charging_efficiency)
+
+    if charging_time > trip_data["Dwell time (hour decimal)"]:
+        charging_time = trip_data["Dwell time (hour decimal)"]
+
+    return charging_time
+
+
+def get_location(location_strategy, dwell_location):
+    """Determines if the vehicle can be charged given location strategy and dwelling location
+
+    :param int location_strategy: where the vehicle can charge-1, 2, 3, 4, or 5; 
+        1-home only, 2-home and work related, 3-anywhere if possibile, 
+        4-home and school only, 5-home and work and school.
+    :param int dwell_location: location the vehicle dwells
+    :return: (*bool*) -- a boolean that represents whether or not the vehicle can charge
+    """
+    # only home
+    if location_strategy == 1:
```

One way this could be refactored is to use a dictionary:
```python
location_allowed = {
    1: dwell_location == 1,  # only home
    2: dwell_location < 13,  # home and go to work related
    3: True,  # anywhere
    4: dwell_location in {1, 21},  # home and school only
    5: dwell_location in {1, 11, 12, 21},  # home and work and school
}
return location_allowed[location_strategy]
```
Besides being more compact, this has the advantage of raising an error if we give an unexpected strategy number, rather than silently returning `None`.